Lombiq.Tests.UI/Extensions/StringExtensions.cs

```diff
@@ -1,12 +1,62 @@
+using System;
 using System.IO;
+using System.Linq;
 
 namespace Lombiq.Tests.UI.Extensions;
 
 public static class StringExtensions
 {
+    private static readonly Lazy<char[]> _invalidPathCharacters = new(() =>
+    new[]
+    {
+        '\0',
+        '\u0001',
+        '\u0002',
+        '\u0003',
+        '\u0004',
+        '\u0005',
+        '\u0006',
+        '\a',
+        '\b',
+        '\t',
+        '\n',
+        '\v',
+        '\f',
+        '\r',
+        '\u000e',
+        '\u000f',
+        '\u0010',
+        '\u0011',
+        '\u0012',
+        '\u0013',
+        '\u0014',
+        '\u0015',
+        '\u0016',
+        '\u0017',
+        '\u0018',
+        '\u0019',
+        '\u001a',
+        '\u001b',
+        '\u001c',
+        '\u001d',
+        '\u001e',
+        '\u001f',
+        '|',
+        ':',
+        '"',
+        '<',
+        '>',
+        '*',
+        '?',
+    }
+        .Concat(Path.GetInvalidFileNameChars())
+        .Concat(Path.GetInvalidPathChars())
+        .Distinct()
```

Comment:
Because they return different values on different OSes. Specifically, `actions/upload-artifact@v2` throws an error because `:` is not stripped from the failure dump path on Linux if you only use `Path.GetInvalidFileNameChars()` and `Path.GetInvalidPathChars()`. It's actually a valid path character (at least on ext4) so that's only a problem when you try to copy the resulting directory to an NTFS drive. 

Strange, I was sure I have added a comment explaining it, maybe it didn't get saved. I wrote a new one:

> The first array contains every invalid character in Windows. This is the union of GetInvalidFileNameChars and
GetInvalidPathChars on Windows (for FAT and NTFS). Linux file systems aren't as strict, they only forbid forward
slash ('/') and null character ('\0'). This is a subset of the invalid characters on Windows. So to make the
paths work across all systems (e.g. when we plan to back up to an NTFS store or use GitHub's
"actions/upload-artifact" action) we have to carry the banned Windows characters across operating systems.
GetInvalidFileNameChars and GetInvalidPathChars are still included in case they contain more characters on
untested operating systems.

Reply:
I remember that discussion now. Thanks again for the insight - the comment lays it out quite nicely.